library/ns1_zone.py

```diff
@@ -568,12 +596,26 @@ def present(self, zone):
         :type zone: dict, optional
         :return: Tuple in which first value reflects whether or not a change
         occured and second value is new or updated zone object
-        :rtype: tuple(bool, dict)
+        :rtype: tuple(bool, dict, dict)
         """
         want = self.sanitize_params(self.module.params)
         if zone:
+            # only if zone already exists
             return self.update_on_change(zone, want)
-        return True, self.create(want)
+        else:
+            # only if zone does not exist
+            for param, value in self.module.params.items():
+                if param not in want:
+                    if value is not None:
+                        want[param] = value
+                else:
+                    if want[param] != value:
+                        want[param] = value
```

Comment:
unfortunately i think comparison to params needs to be done recursively, or another approach, as `value` can be a complex object (dictionary). The create secondary zone integration is failing, as the `secondary` param's value is an object, and _its_ uninitialized values are None, and our API is complaining about it.

this pattern seems applicable to records/other stuff, so it would make sense to me to factor this out into a common function.

Reply:
I took a whack - haven't really tested the corners, but maybe it's a start http://ix.io/2O0T/diff

Reply:
(i should have used `isinstance` instead of `type`)

Reply:
Thank you @rupa! I appreciate the due diligence, you are correct and that is something I didn't notice. Your code looks good, I'll go through the proposed changes and make an update to this MR.